docs/0000-parenting.md

```diff
@@ -0,0 +1,83 @@
+---
+Start Date: 2020-08-04
+RFC PR: (leave this empty)
+Foam Issue: (leave this empty)
+---
+[[index]]
+
+# 0000 Parenting (Notes Hierarchy)
+
+## Summary
+
+I suggest implementing a concept of parenting (visual notes hierarchy): when every note can be a parent and/or a child and the same notes may appear in different parent/children trees. There may be a special children view (similar to backlinks explorer) and some special hierarchy visualisation.
+In general, this is a convenient folders system replacement (one note can be child of several parents).
+
+## Basic example
+
+This concept is brilliantly implemented in [Supernotes](https://supernotes.app) and [Trilium Notes](https://github.com/zadam/trilium) (for example). 
+Imagine set of notes collecting quotes from favorite books:
+``` 
+Author Name
+Another Author Name
+Book Title 1
+Book Title 2
+Book Title 3
+Quote 1
+Quote 2
+Quote 3
+Quote 4
+Quote 5
+Quote 6
+``` 
+In flat notes list these notes are looking like a mess. With parenting they may look like this:
+``` 
+Author Name
+|-- Book Title 1
+    |-- Quote 1
+    |-- Quote 2
+|-- Book Title 2
+    |-- Quote 3
+Another Author Name
+|-- Book Title 3
+    |-- Quote 4
+    |-- Quote 5
+    |-- Quote 6
+``` 
+`Author Name` and `Another Author Name` are roots. There is smth like `parent: Author Name` in the metadata section of `Book Title 1` and `Book Title 2`. There is `parent: Book Title 1` in the metadata section of `Quote 1` and `Quote 2`. There is `parent: Book Title 2` in the metadata section of `Quote 3` etc.
+
+It should be possible to add more than one parent to a note: `parent: Software, Programming`

```

Comment:
I feel like that multiple parentage would be better described as [[links]] or as #tags. We know from programming languages that multiple inheritance gets messy quickly, and that trees are simpler than graphs.

For simplicity, I'd propose that parenting allows every note to have only one parent. That will make visualisation significantly easier, too.

Reply:
Hmm... I fully understand that implementing multiple parents is much more difficult. But having multiple parents seems essential to me: this is one of the main benefits of parenting comparing to a plain file/folder hierarchy, where one note can reside in one and only physical location.